### PR TITLE
WIP load strings dynamically

### PR DESCRIPTION
**Pull request by sgrimm**

### CODE REVIEW

File: src/providers/LocalizationProvider.tsx [hunk 1]
```diff
@@ -32,6 +41,32 @@ export default function LocalizationProvider({ children, locale }: LocalizationP
...
+        // Default to English
+        stringsModule = import('../strings/strings-en');
+      }
+
+      const stringsTable = (await stringsModule).strings;
+
+      const localeMap: ILocalizedStringsMap = {};
+      localeMap[locale] = stringsTable;
+      strings.setContent(localeMap);
+      strings.setLanguage(locale);
```
Comment (karthikbtf) [inline src/providers/LocalizationProvider.tsx:62]: this is great!

**Review by karthikbtf — COMMENTED** [review]
Exciting!!

File: src/components/seeds/database/columns.ts [hunk 2]
```diff
@@ -1,266 +1,268 @@
-import strings from 'src/strings';
+import strings, { ILocalizedStrings } from 'src/strings';
 import { DatabaseColumn } from '@terraware/web-components/components/table/types';
 
-const COLUMNS: DatabaseColumn[] = [
+export type LocalizableDatabaseColumn = Omit<DatabaseColumn, 'name'> & { code: keyof ILocalizedStrings };
+
+const COLUMNS: LocalizableDatabaseColumn[] = [
```
Comment (karthikbtf) [inline src/components/seeds/database/columns.ts:6]: Optionally,, move this into the component scope so it's evaluated on render time.
Another option, convert this into a function that returns the list of columns, so strings are evaluate when the function is called.

Comment (sgrimm) [reply src/components/seeds/database/columns.ts:6]: Converting to a function seems like a good way to go, yeah. That'll also make it easy to memoize.

Comment (sgrimm) [reply src/components/seeds/database/columns.ts]: Updated this to use a function, which definitely seems cleaner. Thanks for that suggestion. It's https://github.com/terraware/terraware-web/pull/1031/commits/6dcd78b75b83c1674fca7ebee0350935c30dcc23 if you want to see it in isolation.

Comment (karthikbtf) [reply src/components/seeds/database/columns.ts]: This looks good, thanks @sgrimm

Comment (sgrimm) [reply src/components/seeds/database/columns.ts]: Updated all the other tables too. Their column lists are all way shorter than the accessions table so it seemed cleaner to inline them into the render functions. (In some cases, they were already inlined and I didn't have to change anything.)

File: src/components/Species/index.tsx [hunk 3]
```diff
@@ -152,8 +153,13 @@ export default function SpeciesList({ reloadData, species }: SpeciesListProps):
       return '';
     }
   };
-  const columns: TableColumnType[] = React.useMemo(
-    () => [
+  const columns: TableColumnType[] = React.useMemo(() => {
```
Comment (karthikbtf) [inline src/components/Species/index.tsx:156]: We could remove the useMemo and see if it really affects performance.
Another option is to maybe pass the strings into the useMemo function - expose the strings in useLocalization, that perhaps negates the benefit of loading strings from `/src/strings`.
For now this is harmless and works.

Comment (sgrimm) [reply src/components/Species/index.tsx:156]: Removing `useMemo` makes the compiler complain that the `useEffect` call that declares `columns` as a dependency will be flagged as changed on every render.

Comment (karthikbtf) [reply src/components/Species/index.tsx:156]: Oops, yes that won't work.

**Comment by sgrimm** [comment]
The changes here have either been merged or are in non-draft PRs, so no more need for this.

**Pull request closed without merging**